File: src/plugin/combat/magic/lunar/teleport/TrollheimTeleportSpell.java
package plugin.combat.magic.lunar.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/28/2017
 */
public class TrollheimTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 92;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(ASTRAL_RUNE, 3, LAW_RUNE, 3, WATER_RUNE, 10);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2807, 3678, 0);
	}
	
	@Override
	public int spellId() {
		return 75;
	}
	
	@Override
	public double exp() {
		return 101;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.LUNAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/TrollheimTeleportSpell.kt
package plugin.interaction.combat.magic.lunar.teleport

import game.content.plugin.combat.spell.type.TeleportSpellPlugin
import game.global.WorldTile
import utility.constants.MagicConstants
import utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/28/2017
 */
class TrollheimTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 92
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.ASTRAL_RUNE, 3, MagicConstants.LAW_RUNE, 3, MagicConstants.WATER_RUNE, 10)
    }

    override fun destination(): WorldTile {
        return WorldTile(2807, 3678, 0)
    }

    override fun spellId(): Int {
        return 75
    }

    override fun exp(): Double {
        return 101.0
    }

    override fun book(): MagicBook {
        return MagicBook.LUNAR
    }
}